src/main/java/ui/UI.java

```diff
@@ -239,6 +242,28 @@ private void initUI(Stage stage) {
         loadFonts();
         String css = initCSS();
         applyCSS(css, scene);
+
+        setApplicationIcon(stage);
+    }
+
+    public void setApplicationIcon(Stage stage) {
+        stage.getIcons().add(new Image(UI.class.getResourceAsStream("logo.png")));
+
+        // set Icon for OSX
+        // - need to use Apple Java Extension, using reflection to load the
+        //   class so that HubTurbo is compilable
+        try {
+            Class util = Class.forName("com.apple.eawt.Application");
+            Method getApplication = util.getMethod("getApplication", new Class[0]);
+            Object application = getApplication.invoke(util);
+            Class params[] = new Class[1];
+            params[0] = java.awt.Image.class;
+            Method setDockIconImage = util.getMethod("setDockIconImage", params);
+            setDockIconImage.invoke(application, new ImageIcon(UI.class.getResource("logo.png")).getImage());
+        } catch (Exception e) {
```

Comment:
Maybe we could explicitly check for OS X using one of the platform-specific methods instead. Feels a bit clearer to read. Otherwise, we can be more specific about which exception is expected here.